nova_vm/src/ecmascript/builtins/array_buffer.rs

```diff
@@ -1,24 +1,54 @@
 //! ## [25.1 ArrayBuffer Objects](https://tc39.es/ecma262/#sec-arraybuffer-objects)
 
 mod data;
-
+use self::data::InternalBuffer;
+use super::{ordinary::ordinary_set_prototype_of_check_loop, Array};
 use crate::{
     ecmascript::{
-        abstract_operations::operations_on_objects::get,
+        abstract_operations::{
+            operations_on_objects::get, testing_and_comparison::same_value_non_number,
+        },
         execution::{agent::JsError, Agent, JsResult},
-        types::{DataBlock, Function, InternalMethods, Number, Object, PropertyKey, Value},
+        types::{
+            DataBlock, Function, InternalMethods, Number, Object, OrdinaryObject,
+            OrdinaryObjectInternalSlots, PropertyDescriptor, PropertyKey, Value,
+        },
     },
     heap::{indexes::ArrayBufferIndex, GetHeapData},
     Heap,
 };
-
 pub use data::ArrayBufferHeapData;
-
-use self::data::InternalBuffer;
+use std::ops::Deref;
 
 #[derive(Debug, Copy, Clone, Eq, PartialEq)]
 pub struct ArrayBuffer(ArrayBufferIndex);
 
+impl From<ArrayBufferIndex> for ArrayBuffer {
+    fn from(value: ArrayBufferIndex) -> Self {
+        ArrayBuffer(value)
+    }
```

Comment:
Yeah, defining all the `From<_>` for something is definitely better at the target level. It beats having a ton of `From<_> for Value` in the `value.rs` file.